### PR TITLE
Clean test runner up

### PR DESCRIPTION
**Pull request by seunlanlege**
This just bumps `parity-scale-codec` in test-runner to `v2`

### CODE REVIEW

File: test-utils/test-runner/Cargo.toml [hunk 1]
```diff
@@ -43,7 +43,7 @@ sp-runtime-interface = { version = "3.0.0", path = "../../primitives/runtime-int
 # pallets
 frame-system = { version = "3.0.0", path = "../../frame/system" }
 
-parity-scale-codec = "1.3.1"
+parity-scale-codec = "2.0.0"
 env_logger = "0.7.1"
 log = "0.4.8"
 futures01 = { package = "futures", version = "0.1.29" }
```
Comment (ordian) [inline test-utils/test-runner/Cargo.toml:49]: is it actually used? what about futures01 and rand?

Comment (bkchr) [reply test-utils/test-runner/Cargo.toml:49]: Scale-codec is also not used anywhere.

Comment (seunlanlege) [reply test-utils/test-runner/Cargo.toml]: rand is used, but not futures01

**Comment by seunlanlege** [comment]
@ordian @bkchr please take a look at this again

File: test-utils/test-runner/Cargo.toml (hunk 1 above)
Comment (ordian) [reply test-utils/test-runner/Cargo.toml]: rand in used here
https://github.com/paritytech/substrate/blob/aa3344161522648039fd1080f03396775d1a13bf/test-utils/test-runner/src/node.rs#L160
but is it actually needed?
https://github.com/libp2p/rust-libp2p/blob/bb206bcfdb8f43b71a560cc6938080369b033d48/misc/multiaddr/src/protocol.rs#L80

**Review by ordian — APPROVED** [review]

File: test-utils/test-runner/Cargo.toml [hunk 2]
```diff
@@ -43,18 +43,11 @@ sp-runtime-interface = { version = "3.0.0", path = "../../primitives/runtime-int
 # pallets
 frame-system = { version = "3.0.0", path = "../../frame/system" }
 
-parity-scale-codec = "1.3.1"
 env_logger = "0.7.1"
 log = "0.4.8"
-futures01 = { package = "futures", version = "0.1.29" }
 futures = { package = "futures", version = "0.3", features = ["compat"] }
 rand = "0.7"
```
Comment (ordian) [inline test-utils/test-runner/Cargo.toml:49]:
```suggestion
```

**Comment by seunlanlege** [comment]
hey @kianenigma @gnunicorn could you please take a look at this?

File: test-utils/test-runner/src/utils.rs [hunk 3]
```diff
@@ -20,10 +20,15 @@ use futures::{Sink, SinkExt};
...
+use sc_service::BasePath;
 
 /// Base db path gotten from env
-pub fn base_path() -> Option<String> {
-	std::env::var("DB_BASE_PATH").ok()
+pub fn base_path() -> BasePath {
+	if let Some(base) = std::env::var("DB_BASE_PATH").ok() {
+		BasePath::new(base)
+	} else {
+		BasePath::new_temp_dir().expect("couldn't create a temp dir")
```
Comment (kianenigma) [inline test-utils/test-runner/src/utils.rs:30]: Isn't this like a random path?

Comment (seunlanlege) [reply test-utils/test-runner/src/utils.rs:30]: yeah, it is, if you don't supply an existing db path, it is assumed you want to test with a new db

**Comment by seunlanlege** [comment]
bump

**Comment by seunlanlege** [comment]
bump

File: test-utils/test-runner/src/utils.rs (hunk 3 above)
Comment (kianenigma) [reply test-utils/test-runner/src/utils.rs:30]: yeah okay I was wondering if loading a fresh DB would make any sense, but yeah I presume you could also start with a new DB and it should be fine.

File: bin/node/test-runner-example/src/lib.rs [hunk 4]
```diff
@@ -71,6 +81,89 @@ impl ChainInfo for NodeTemplateChainInfo {
 		)
 	}
 
+	fn config(task_executor: TaskExecutor) -> Configuration {
```
Comment (kianenigma) [inline bin/node/test-runner-example/src/lib.rs:84]: why can't this stay in `utils/test-runner`? I was happy that in the previous PR you made it such, but why is it being reverted now? 

I still think that 80% of the things being configured here are pretty irrelevant to the end user/programmer, thus not a good API.

Comment (kianenigma) [reply bin/node/test-runner-example/src/lib.rs:84]: At least we can provide different variations where you have `fn config()` as it is now, and a `config_default()` or something like this.

Comment (seunlanlege) [reply bin/node/test-runner-example/src/lib.rs:84]: There are cases where we'd like to run the test runner as a bin and provide it with command line args like a real node. (This is all simnet stuff), yeah i suppose a default would be nice 🤔

**Review by gnunicorn — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]

**Pull request merged**